python/TestHarness/util.py

```diff
@@ -379,17 +379,17 @@ def getCompilers(libmesh_dir):
     if "distcc" in mpicxx_cmd or "ccache" in mpicxx_cmd:
         mpicxx_cmd = mpicxx_cmd.split()[-1]
 
-    # If mpi ic on the command, run -show to get the compiler
+    # If mpi is in the command, run -show to get the compiler
     if "mpi" in mpicxx_cmd:
         raw_compiler = runCommand(mpicxx_cmd + " -show")
     else:
         raw_compiler = mpicxx_cmd
 
-    if re.match('icpc', raw_compiler) != None:
+    if re.match('\S*icpc\s', raw_compiler) != None:
         compilers.add("INTEL")
-    elif re.match('[cg]\+\+', raw_compiler) != None:
+    elif re.match('\S*[cg]\+\+\s', raw_compiler) != None:
```

Comment:
This will match both g++ and clan[g]++

That is why I found it necessary to use a negative look behind operation (in my previous PR attempt). This seems to work:

```diff
-elif re.match('\S*[cg]\+\+\s', raw_compiler) != None:
+elif re.match('\S*(?<!clan)[cg]\+\+\s', raw_compiler) != None:
```

Reply:
Sure - but a better solution would be to reverse these last to branches so that the clang check comes first. Then you'll know that the GCC check won't match clang without the extra complication.

Reply:
will do